src/js/src/components/ToolboxComponents/DomainStats.vue

```diff
@@ -3,22 +3,114 @@
+        Generate
+      </button>
+      <div v-if="loading === true" class="spinner" />
+      <div v-show="loading === false && rawData" id="lineContainer">
+        <canvas
+          id="line-chart"
+          width="800"
+          height="450" />
+      </div>
+      <div v-if="rawData !== null && loading === false" id="tableContainer">
+        <table id="domainGrowthTable">
+          <thead>
+            <tr>
+              <th />
+              <th v-for="(item, index) in rawData" :key="index">
+                {{ item.year }}
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>Size in KB</td>
+              <td v-for="(item, index) in rawData" :key="index">
+                {{ item.sizeInKb.toLocaleString("en") }}
+              </td>
+            </tr>
+            <tr>
+              <td>Total pages</td>
+              <td v-for="(item, index) in rawData" :key="index">
+                {{ item.totalPages.toLocaleString("en") }}
+              </td>
+            </tr>
+            <tr>
+              <td>Ingoing links</td>
+              <td v-for="(item, index) in rawData" :key="index">
+                {{ item.ingoingLinks.toLocaleString("en") }}
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    </div>
   </div>
 </template>
 
 <script>
 
+import { requestService } from '../../services/RequestService'
+import domainScript from './ToolboxResources/domainStats'
+
 export default {
   name: 'DomainStats',
   data() {
     return {
+      domain:'',
+      loading: false,
+      rawData:null,
+      graphData:{
+        chartLabels:[],
+        sizeInKb:[],
+        ingoingLinks:[],
+        numberOfPages:[]
+      }
     }
   },
   mounted () {
-
   },
   methods: {
+    loadGraphData(domain) {
+      this.graphData = {
+        chartLabels:[],
+        sizeInKb:[],
+        ingoingLinks:[],
+        numberOfPages:[]
+      }
+      this.rawData = null
+      this.loading = true
+      requestService.getDomainStatistics(this.prepareDomainForGetRequest()).then(result => (this.sanitizeResponseDataAndDrawChart(result), this.rawData = result), error => console.log('No information found about this archive.'))
+    },
+    checkDomain(domain) {
+      return true
+    },
+    prepareDomainForGetRequest() {
+      let preparedDomain = this.domain
+      preparedDomain = preparedDomain.replace(/http.*:\/\//i,'').trim() //Get domain from URL, using replace and regex to trim domain
+      preparedDomain = preparedDomain.replace(/www./i,'') //Get domain from URL, using replace and regex to trim domain
+      if( preparedDomain.slice(-1) === '/'){ // if trailing slash on domain it's removed
+        preparedDomain = preparedDomain.slice(0, -1)
+      }
+      return preparedDomain
+    },
+    sanitizeResponseDataAndDrawChart(data) {
+      for(var i = 0; i < data.length; i++){
```

Comment:
Good ol' _for_ loop! - miss those - so very nice to bump into legacy code :) Consider changing _var_ to _let_ just to keep it consistent when we are in Vue components land. It's fine to keep _var_ and all other legacy stuff intact when separated out in .js files.